parachain/pallets/eth-app/src/lib.rs

```diff
@@ -152,13 +155,12 @@ pub mod pallet {
 			let recipient = T::Lookup::lookup(recipient)?;
 			T::Asset::deposit(&recipient, amount)?;
 
-			if let Some(id) = para_id {
+			if let Some(destination) = destination {
 				T::XcmReserveTransfer::reserve_transfer(
```

Comment:
We're going to lose the user's funds here. The reserve transfer should be attempted last, after the `deposit_event`. If the function returns an error, it should be ignored (though we should log the error at the very least).

If we don't do the above, then a failed reserve transfer would cause `mint` to fail and the storage transaction for `mint` to be rolled back (as denoted by that `#[transactional]` annotation). This would rollback the `T::Asset::deposit()` and the user would lose their funds.

Reply:
yeh - again making sure we have a document that covers every possible scenario/edge case and ensures we have both an acceptable on-chain outcome as well as ui/ux outcome that preserves user funds irrespective of the scenario is important.

Reply:
Changed this to log error on failure and then continue.